vulcan/lib/server.rb

```diff
@@ -12,25 +13,27 @@ class Server < Etna::Server
     end
 
     get 'api/workflows', action: 'workflows#fetch', as: :workflows_view, auth: { user: { active?: true, has_flag?: 'vulcan' } }
-    get 'api/:project_name/data/:cell_hash/:data_filename', action: 'data#fetch', as: :data_view, match_ext: true, auth: { user: { can_view?: :project_name, has_flag?: 'vulcan' } }
-    post 'api/:project_name/session/:workflow_name/status', action: 'sessions#status', as: :status_view, match_ext: true, auth: { user: { can_view?: :project_name, has_flag?: 'vulcan' } }
-    post 'api/:project_name/session/:workflow_name', action: 'sessions#submit', as: :submit_view, match_ext: true, auth: { user: { can_view?: :project_name, has_flag?: 'vulcan' } }
 
-    with auth: { user: { active?: true, has_flag?: 'vulcan' } } do
+    with auth: { user: { can_view?: :project_name, has_flag?: 'vulcan' } } do
+      get 'api/:project_name/data/:cell_hash/:data_filename', action: 'data#fetch', as: :data_view, match_ext: true
+      post 'api/:project_name/session/:workflow_name/status', action: 'sessions#status', as: :status_view, match_ext: true
+      post 'api/:project_name/session/:workflow_name', action: 'sessions#submit', as: :submit_view, match_ext: true
+
+      get 'api/:project_name/figures', action: 'figure#fetch'
+      get 'api/:project_name/figure/:figure_id', action: 'figure#get'
+      post 'api/:project_name/figure/create', action: 'figure#create'
+      post 'api/:project_name/figure/:figure_id/update', action: 'figure#update'
+    end
 
-      # remaining view routes are parsed by the client and must also be set there
-      get '/:project_name/workflow', as: :workflow do
-        erb_view(:client)
-      end
+    with auth: { user: { active?: true, has_flag?: 'vulcan' } } do
```

Comment:
These can be grouped with the previous `auth` block, I think, and consolidated under the `user: can_view` instead of `user: active`?